Makefile

```diff
@@ -29,8 +29,16 @@ VERSION_LABEL ?= ${VERSION}
 # The go commands and the minimum Go version that must be used to build the app.
 GO ?= go
 GOFMT ?= $(shell ${GO} env GOROOT)/bin/gofmt
+GO_MOD_VERSION = $(shell sed -En 's/^go[[:space:]]+([[:digit:].]+)/\1/p' go.mod)
 GO_VERSION_KIALI = 1.22.1
 GO_TEST_FLAGS ?=
+all: check_version
+
+check_version:
+	@if [ "$(GO_MOD_VERSION)" != "$(GO_VERSION_KIALI)" ]; then \
+		echo "Kiali Go version ${GO_VERSION_KIALI} different than go.mod ${GO_MOD_VERSION}"; \
+		exit 1; \
+	fi
```

Comment:
You will want to get rid of that `all` target, and move the check code into the `go-check` target found in `make/Makefile.build.mk` here: https://github.com/kiali/kiali/blob/master/make/Makefile.build.mk#L24-L28